tests/exception_handling/synchronous_exceptions.cpp

```diff
@@ -8,11 +8,90 @@
 
 #include "../common/common.h"
 
+#include <limits>
+
 #define TEST_NAME synchronous_exceptions
 
 namespace TEST_NAMESPACE {
 using namespace sycl_cts;
 
+class custom_exception : public sycl::exception {
+ public:
+  struct reference {
+    static std::string what() {
+      return "custom_exception";
+    }
+    static std::error_code code() {
+      return sycl::make_error_code(sycl::errc::runtime);
+    }
+    static const std::error_category& category() {
+      return sycl::sycl_category();
+    }
+  };
+  custom_exception(sycl::context ctx) :
+      sycl::exception(ctx, reference::code(), reference::what()) {}
+};
+
+/** @brief Verify exception propagation from command group
+ */
+void check_exception_api(util::logger &log) {
+  auto q = util::get_cts_object::queue();
+  auto ctx = q.get_context();
+
+  try {
+    q.submit([&](cl::sycl::handler &cgh) {
+      cgh.single_task<class TEST_NAME>([=]() {});
+
+      throw custom_exception(ctx);
+    });
+    q.wait_and_throw();
+  } catch (const custom_exception &e) {
```

Comment:
Just curious here...
Are you expecting that the exception is exiting the `submit` or is just captured and propagated to the `q.wait_and_throw()`?
My interpretation of the standard is that the exception should exit directly from the `submit`.
So, could we just replace `q.wait_and_throw()` by `q.wait()` or even remove the wait at all?

Reply:
You are right, it's better to remove the blocking wait. Done by 214192da31bb